test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -53,31 +53,69 @@ struct test_inclusive_scan_by_segment
         }
     }
 
-    template <typename Iterator1, typename Iterator2, typename Size>
+    template <typename Iterator>
+    int
+    accum_values(Iterator host_vals, int idx_begin, int idx_end)
```

Comment:
I would recommend to use std::accumulate here.

Reply:
Added cash into accum_values for avoid quadratic complexity.

Reply:
> I would recommend to use std::accumulate here.
Fixed